jsapp/js/components/RESTServices/RESTServicesForm.tsx

```diff
@@ -356,22 +400,25 @@ export default class RESTServicesForm extends React.Component {
                 id={`headerValue-${n}`}
                 name='headerValue'
                 value={this.state.customHeaders[n].value}
-                data-index={n}
-                onChange={this.handleCustomHeaderChange.bind(this)}
-                onKeyPress={this.onCustomHeaderInputKeyPress.bind(this)}
+                onChange={(evt: React.ChangeEvent<HTMLInputElement>) => {
+                  this.handleCustomHeaderValueChange(n, evt.target.value);
+                }}
+                onKeyDown={this.onCustomHeaderInputKeyPress.bind(this)}
               />
 
               <Button
                 type='secondary-danger'
                 size='m'
                 className='http-header-row-remove'
-                data-index={n}
                 startIcon='trash'
-                onClick={this.removeCustomHeaderRow.bind(this)}
+                onClick={(evt) => {
```

Comment:
like above, beware of typos in the `evt.preventDefault();`, now it's type-safe :)
```suggestion
                onClick={(evt: React.ChangeEvent<HTMLInputElement>) => {
```